### PR TITLE
[DependencyInjection] Document BC break in UPGRADE file

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | Follows #52843
| License       | MIT

### CODE REVIEW

File: UPGRADE-7.0.md [hunk 1]
```diff
@@ -145,6 +145,7 @@ DependencyInjection
    To fetch services lazily, you can use a [service subscriber](https://symfony.com/doc/6.4/service_container/service_subscribers_locators.html#defining-a-service-subscriber).
  * Add parameter `string $id = null` and `bool &$asGhostObject = null` to `LazyProxy\PhpDumper\DumperInterface::isProxyCandidate()` and `getProxyCode()`
  * Add parameter `string $source = null` to `FileLoader::registerClasses()`
+ * [BC BREAK] When used in the `prependExtension()` methods, the `ContainerConfigurator::import()` method now prepends the configuration instead of appending it
```
Comment (yceruto) [inline UPGRADE-7.0.md]:
```
`prependExtension()` method 
```
in singular?

**Review by yceruto — COMMENTED** [review]
Oskar shouldn't the file to be updated be `UPGRADE-7.1.md` instead?

File: UPGRADE-7.0.md (hunk 1 above)
Comment (yceruto) [reply UPGRADE-7.0.md]: Same for the DI/CHANGELOG.md ?

**Comment by OskarStark** [comment]
You are right, I picked the wrong one 😅

File: UPGRADE-7.0.md (hunk 1 above)
Comment (OskarStark) [reply UPGRADE-7.0.md]: Will fix it

**Review by yceruto — APPROVED** [review]
Thanks Oskar!

**Review by xabbuh — APPROVED** [review]

**Comment by fabpot** [comment]
Sorry, I'm late in the game here, but what is the rationale behind doing a BC break in 7.1?

**Comment by yceruto** [comment]
> Sorry, I'm late in the game here, but what is the rationale behind doing a BC break in 7.1?

Hey Fabien, we published this as a BC break because technically it is (we changed the behavior of the `import()` method under certain context/scope without deprecation path). However, actually, nobody can/should use the `import()` method in the `prependExtension()` because it's defective, appending config instead of prepending. That's why we were using the `ContainerBuilder::prependExtensionConfig()` until now (which is limited to an `array` of config).

Honestly, this BC break, in the end, doesn't feel like a real BC break to me, as the previous behavior was faulty and unwanted. However, it applies to the BC break rules, so we are documenting it as such.

More details can be found here https://github.com/symfony/symfony/issues/52789 and here https://github.com/symfony/symfony/pull/52843

Hope it's ok for you🤞

**Comment by fabpot** [comment]
> > Sorry, I'm late in the game here, but what is the rationale behind doing a BC break in 7.1?
> 
> Hey Fabien, we published this as a BC break because technically it is (we changed the behavior of the `import()` method under certain context/scope without deprecation path). However, actually, nobody can/should use the `import()` method in the `prependExtension()` because it's defective, appending config instead of prepending. That's why we were using the `ContainerBuilder::prependExtensionConfig()` until now (which is limited to an `array` of config).
> 
> Honestly, this BC break, in the end, doesn't feel like a real BC break to me, as the previous behavior was faulty and unwanted. However, it applies to the BC break rules, so we are documenting it as such.
> 
> More details can be found here #52789 and here #52843
> 
> Hope it's ok for you🤞

Thank you for the clear explanation, that makes sense to me now.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @OskarStark.

**Pull request merged**